test/EventBasedPredictionMarket.ts

```diff
@@ -0,0 +1,181 @@
+import { OptimisticOracle } from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers";
+import { EventBasedPredictionMarket, ExpandedERC20 } from "../typechain";
+import { amountToSeedWallets } from "./constants";
+import { eventBasedPredictionMarketFixture } from "./fixtures/EventBasedPredictionMarket.Fixture";
+import { umaEcosystemFixture } from "./fixtures/UmaEcosystem.Fixture";
+import { BigNumber, Contract, ethers, expect, SignerWithAddress, toWei } from "./utils";
+
+let eventBasedPredictionMarket: EventBasedPredictionMarket, usdc: Contract;
+let optimisticOracle: OptimisticOracle;
+let longToken: ExpandedERC20;
+let shortToken: ExpandedERC20;
+let deployer: SignerWithAddress, sponsor: SignerWithAddress, holder: SignerWithAddress, disputer: SignerWithAddress;
+
+describe("EventBasedPredictionMarket functions", function () {
+  const proposeAndSettleOptimisticOraclePrice = async (price: BigNumber) => {
+    const ancillaryData = await eventBasedPredictionMarket.customAncillaryData();
+    const identifier = await eventBasedPredictionMarket.priceIdentifier();
+    const expirationTimestamp = await eventBasedPredictionMarket.expirationTimestamp();
+    const optimisticOracleLivenessTime = await eventBasedPredictionMarket.optimisticOracleLivenessTime();
+
+    await optimisticOracle
+      .connect(deployer)
+      .proposePrice(eventBasedPredictionMarket.address, identifier, expirationTimestamp, ancillaryData, price);
+
+    await optimisticOracle
+      .connect(deployer)
+      .setCurrentTime((await optimisticOracle.getCurrentTime()).add(optimisticOracleLivenessTime));
+
+    await optimisticOracle
+      .connect(deployer)
+      .settle(eventBasedPredictionMarket.address, identifier, expirationTimestamp, ancillaryData);
+  };
```

Comment:
nit: as the first address set in the `ethers.getSigners()` _is_ the `deployer`, you don't need to do the `.connect` syntax. I think where possible we should try keep this as 1-liners for each function call. these tests are _very_ simple and so if we can get there to be no multi-line calls like this it would help with keeping it "looking simple".